### PR TITLE
Dcandeau/1622/no edit message

### PR DESCRIPTION
**Pull request by davidcandeau**
## 🍂 Before You Submit

* [ ] Did you write & run tests and linters?
* [ ] Did you run Google Lighthouse and/or WebAIM (Wave) on UI components in your PR?
* [ ] Does your PR contain any required translations?
* [ ] Are your primary keys UUIDs on any new tables?

## 🛷 Deployment

* [ ] Migrations
* [ ] Dependencies

### CODE REVIEW

**Review by katelovescode — CHANGES_REQUESTED** [review]
My initial instinct is that the trigger to check if a payment has been recorded for this month yet belongs in the paymentModal component itself.  Given that eventually we'll have a drop down to select prior months, I'm not sure if there's a practical way to store "isPaymentSubmitted" from the response of submitting the payment; you might just need to make the request every time the modal opens.  

As implemented now, the page requires a refresh to display the "already submitted" message - i.e. if I submit a payment, click okay on the success message, then click "record new > payment" again, I can submit more payments.

@cjhaddad what are your thoughts on this?

**Comment by cjhaddad** [comment]
> My initial instinct is that the trigger to check if a payment has been recorded for this month yet belongs in the paymentModal component itself. Given that eventually we'll have a drop down to select prior months, I'm not sure if there's a practical way to store "isPaymentSubmitted" from the response of submitting the payment; you might just need to make the request every time the modal opens.
> 
> As implemented now, the page requires a refresh to display the "already submitted" message - i.e. if I submit a payment, click okay on the success message, then click "record new > payment" again, I can submit more payments.
> 
> @cjhaddad what are your thoughts on this?



> My initial instinct is that the trigger to check if a payment has been recorded for this month yet belongs in the paymentModal component itself. Given that eventually we'll have a drop down to select prior months, I'm not sure if there's a practical way to store "isPaymentSubmitted" from the response of submitting the payment; you might just need to make the request every time the modal opens.
> 
> As implemented now, the page requires a refresh to display the "already submitted" message - i.e. if I submit a payment, click okay on the success message, then click "record new > payment" again, I can submit more payments.
> 
> @cjhaddad what are your thoughts on this?

Yes, it looks like it definitely makes sense to put the call to check if a payment is submitted needs to occur each time the Modal opens. The behavior we see now is because the call happens once and is stored in the state but not updated again after the payment modal closes and has successful submission.

**Comment by cjhaddad** [comment]
> My initial instinct is that the trigger to check if a payment has been recorded for this month yet belongs in the paymentModal component itself. Given that eventually we'll have a drop down to select prior months, I'm not sure if there's a practical way to store "isPaymentSubmitted" from the response of submitting the payment; you might just need to make the request every time the modal opens.
> 
> As implemented now, the page requires a refresh to display the "already submitted" message - i.e. if I submit a payment, click okay on the success message, then click "record new > payment" again, I can submit more payments.
> 
> @cjhaddad what are your thoughts on this?



> My initial instinct is that the trigger to check if a payment has been recorded for this month yet belongs in the paymentModal component itself. Given that eventually we'll have a drop down to select prior months, I'm not sure if there's a practical way to store "isPaymentSubmitted" from the response of submitting the payment; you might just need to make the request every time the modal opens.
> 
> As implemented now, the page requires a refresh to display the "already submitted" message - i.e. if I submit a payment, click okay on the success message, then click "record new > payment" again, I can submit more payments.
> 
> @cjhaddad what are your thoughts on this?

Yes, it looks like it definitely makes sense to put the call to check if a payment is submitted needs to occur each time the Modal opens. The behavior we see now is because the call happens once and is stored in the state but not updated again after the payment modal closes and has successful submission.

File: client/src/Payment/paymentModal.js [hunk 1]
```diff
@@ -1,19 +1,23 @@
 import React, { useEffect, useState } from 'react'
```
Comment (katelovescode) [inline client/src/Payment/paymentModal.js:1]: @davidcandeau small note for consistency, this file should be named `PaymentModal` - not sure how that slipped past us.

File: app/models/attendance.rb [hunk 2]
```diff
@@ -132,16 +132,16 @@ def delete_or_mark_absent
...
-#  created_at        :datetime         not null
-#  updated_at        :datetime         not null
-#  child_approval_id :uuid             not null
-#  service_day_id    :uuid
-#  wonderschool_id   :string
+#  id                                                       :uuid             not null, primary key
+#  check_in                                                 :datetime         not null
+#  check_out                                                :datetime
+#  deleted_at                                               :date
+#  time_in_care(Calculated: check_out time - check_in time) :interval         not null
```
Comment (katelovescode) [inline app/models/attendance.rb:139]: Don't commit these changes; I'm working to get annotate to stop doing it

File: app/models/nebraska_dashboard_case.rb [hunk 3]
```diff
@@ -27,7 +27,7 @@ class NebraskaDashboardCase < UuidApplicationRecord
 #  full_days_remaining        :integer          default(0), not null
 #  hours                      :float            default(0.0), not null
 #  hours_remaining            :float            default(0.0), not null
-#  month                      :datetime         default(Tue, 10 May 2022 19:40:48.447958000 UTC +00:00), not null
```
Comment (katelovescode) [inline app/models/nebraska_dashboard_case.rb:30]: Ditto here on unrelated changes

File: client/src/Payment/paymentModal.js (hunk 1 above)
Comment (katelovescode) [reply client/src/Payment/paymentModal.js:1]: :boop: on this one, can you rename this file, please?

File: spec/factories/attendances.rb [hunk 4]
```diff
@@ -102,16 +102,16 @@
...
-#  created_at        :datetime         not null
-#  updated_at        :datetime         not null
-#  child_approval_id :uuid             not null
-#  service_day_id    :uuid
-#  wonderschool_id   :string
+#  id                                                       :uuid             not null, primary key
+#  check_in                                                 :datetime         not null
+#  check_out                                                :datetime
+#  deleted_at                                               :date
+#  time_in_care(Calculated: check_out time - check_in time) :interval         not null
```
Comment (katelovescode) [inline spec/factories/attendances.rb:109]: Ditto here, uncommit pls

File: db/schema.rb [hunk 5]
```diff
@@ -191,7 +191,7 @@
   end
 
   create_table "nebraska_dashboard_cases", id: :uuid, default: -> { "gen_random_uuid()" }, force: :cascade do |t|
-    t.datetime "month", default: "2022-05-10 19:40:48", null: false
+    t.datetime "month", default: "2022-05-11 19:17:16", null: false
```
Comment (katelovescode) [inline db/schema.rb:194]: None of this should be committed, also same with pie_erd (I can't comment on that file because it's binary)

**Review by katelovescode — CHANGES_REQUESTED** [review]
Please uncommit the files as requested, but otherwise this works great.  I'll approve once the extraneous files are pulled out.

**Comment by katelovescode** [comment]
Reopening as a separate branch.

**Pull request closed without merging**